src/Bicep.Core/Semantics/Namespaces/ThirdPartyNamespaceType.cs

```diff
@@ -17,14 +18,32 @@ public static class ThirdPartyNamespaceType
 
         public static NamespaceType Create(string name, string aliasName, IResourceTypeProvider resourceTypeProvider)
         {
-            return new NamespaceType(
-                aliasName,
-                new NamespaceSettings(
+            var namespaceSettings = new NamespaceSettings(
                     IsSingleton: true,
                     BicepProviderName: name,
                     ConfigurationType: null,
                     ArmTemplateProviderName: name,
-                    ArmTemplateProviderVersion: resourceTypeProvider.Version),
+                    ArmTemplateProviderVersion: resourceTypeProvider.Version);
+
+            if (resourceTypeProvider is ThirdPartyResourceTypeProvider thirdPartyProvider)
+            {
+                var namespaceConfig = thirdPartyProvider.GetNamespaceConfiguration();
+
+                if (namespaceConfig is not null)
```

Comment:
I think we should just require `namespaceConfig` to be set for 3rd party providers, and just throw an exception if it isn't.